tests/token_token_create_transaction.rs

```diff
@@ -0,0 +1,318 @@
+        .auto_renew_period(used_data.auto_renew_period)
+        .token_memo(&used_data.memo)
+        .token_supply_type(used_data.token_supply_type);
+
+    // Add either one publickey (same as operator) or multiple keys with keylist
+    // Sing transaction with the required number of keys if keylist was used
+    match &used_data.public_key {
+        PublicKeyType::PublicKey(key) => {
+            token_create_tx
+                .admin_key(key.clone())
+                .kyc_key(key.clone())
+                .freeze_key(key.clone())
+                .wipe_key(key.clone())
+                .supply_key(key.clone())
+                .fee_schedule_key(key.clone())
+                .pause_key(key.clone());
+            ();
+        }
+        PublicKeyType::KeyList(keylist) => {
+            token_create_tx
+                .admin_key(keylist.clone())
+                .kyc_key(keylist.clone())
+                .freeze_key(keylist.clone())
+                .wipe_key(keylist.clone())
+                .supply_key(keylist.clone())
+                .fee_schedule_key(keylist.clone())
+                .pause_key(keylist.clone());
+
+            // Signing of the transaction
+            match &used_data.private_keys {
+                    Some(private_keys) => {
+                        match keylist.threshold {
+                            // Sign only with the required number of Privatekey according to keylist threshold
+                            Some(threshold) => {
+                                for i in 0..threshold {
+                                    token_create_tx.sign(private_keys[i as usize].clone());
+                                }
+                                ();
+                            }
+                            // Sign with all Privatekeys
+                            None => {
+                                for private_key in private_keys {
+                                    token_create_tx.sign(private_key.clone());
+                                }
+                                ();
+                            }
+                        }
+                    }
+                    None => panic!("Keylist with public keys but no list with private keys to sign the transaction provided."),
+                }
+            ();
+        }
+    }
+
+    // If token supply is finite a maximum supply must be set
+    if used_data.token_supply_type == TokenSupplyType::Finite {
+        match used_data.max_supply {
+            Some(max_supply) => {
+                token_create_tx.max_supply(max_supply);
+                ();
+            }
+            None => panic!("Token Supply Type is finite but no max supply was provided."),
+        }
+    }
+
+    // Execute transaction with max fee of 100 HBar to be sure it gets executed
+    // Get transaction receipt and return only token id
+    let token_id = token_create_tx
+        .max_transaction_fee(Hbar::new(100))
```

Comment:
This shouldn't be necessary (in fact, if the default isn't enough, we should probably have a bigger default)